e2e/suites/management/delete_scheduler_test.go

```diff
@@ -0,0 +1,125 @@
+
+	"github.com/topfreegames/maestro/e2e/framework/maestro"
+
+	"github.com/topfreegames/maestro/e2e/framework"
+	"k8s.io/client-go/kubernetes"
+)
+
+func TestDeleteScheduler(t *testing.T) {
+	t.Parallel()
+
+	framework.WithClients(t, func(roomsApiClient *framework.APIClient, managementApiClient *framework.APIClient, kubeClient kubernetes.Interface, redisClient *redis.Client, maestro *maestro.MaestroInstance) {
+		t.Run("Should Succeed - check pods and namespace are deleted", func(t *testing.T) {
+			t.Parallel()
+
+			game := "delete-game"
+			scheduler, err := createSchedulerWithRoomsAndWaitForIt(t, maestro, managementApiClient, game, kubeClient)
+			require.NoError(t, err)
+
+			deleteSchedulerRequest := &maestroApiV1.DeleteSchedulerRequest{SchedulerName: scheduler.Name}
+			deleteSchedulerResponse := &maestroApiV1.DeleteSchedulerResponse{}
+			err = managementApiClient.Do("DELETE", fmt.Sprintf("/schedulers/%s", scheduler.Name), deleteSchedulerRequest, deleteSchedulerResponse)
+			assert.NoError(t, err)
+			assert.NotNil(t, deleteSchedulerResponse.OperationId, scheduler.Name)
+
+			// Assert every pod is deleted
+			assert.Eventually(t, func() bool {
+				pods, err := kubeClient.CoreV1().Pods(scheduler.Name).List(context.Background(), metav1.ListOptions{})
+				assert.NoError(t, err)
+				if len(pods.Items) == 0 {
+					return true
+				}
+				return false
+			}, time.Second*60, time.Second)
+
+			// Assert namespace is deleted
+			assert.Eventually(t, func() bool {
+				ns, err := kubeClient.CoreV1().Namespaces().List(context.Background(), metav1.ListOptions{})
+				assert.NoError(t, err)
+
+				for _, n := range ns.Items {
+					if scheduler.Name == n.GetName() {
+						return false
+					}
+				}
+				return true
+			}, time.Second*60, time.Second)
+
+			// Assert scheduler is deleted
+			assert.Eventually(t, func() bool {
+				getSchedulerRequest := &maestroApiV1.GetSchedulerRequest{SchedulerName: scheduler.Name}
+				getSchedulerResponse := &maestroApiV1.GetSchedulerResponse{}
+				err = managementApiClient.Do("GET", fmt.Sprintf("/schedulers/%s", scheduler.Name), getSchedulerRequest, getSchedulerResponse)
+
+				if getSchedulerResponse.GetScheduler() == nil {
+					return true
+				}
+				return false
+			}, time.Second*60, time.Second*5)
+
+			// Assert scheduler operations are deleted
+			assert.Eventually(t, func() bool {
+				getOperationsRequest := &maestroApiV1.ListOperationsRequest{SchedulerName: scheduler.Name}
+				getOperationsResponse := &maestroApiV1.ListOperationsResponse{}
+				err = managementApiClient.Do("GET", fmt.Sprintf("/scheduler/%s/operations", scheduler.Name), getOperationsRequest, getOperationsResponse)
+
+				if len(getOperationsResponse.GetFinishedOperations()) == 0 &&
+					len(getOperationsResponse.GetActiveOperations()) == 0 &&
+					len(getOperationsResponse.GetPendingOperations()) == 0 {
+					return true
+				}
+				return false
+			}, time.Second*60, time.Second*5)
```

Comment:
Hey, @caiordjesus, do you think it is worth trying to create the same scheduler after the deletion to ensure the full cleanup was succeeded?